transformation/src/main/scala/org/broadinstitute/monster/clinvar/Content.scala

```diff
@@ -1,19 +1,128 @@
 package org.broadinstitute.monster.clinvar
 
 import ujson.StringRenderer
-import upack.Msg
+import upack._
+
+import scala.collection.mutable
 
 object Content {
 
   /**
     * Encode unmodeled data so it can be included in a string column.
     *
-    * If the data is empty, it will be dropped.
+    * If the data is empty, it will be dropped. Otherwise, it will be
+    * converted to a canonical form, then encoded with no extra whitespace.
     */
   def encode(msg: Msg): Option[String] =
-    if (msg.obj.isEmpty) {
-      None
+    msg match {
+      case Null                        => None
+      case Obj(kvs) if kvs.isEmpty     => None
+      case Arr(items) if items.isEmpty => None
+      case other                       => Some(upack.transform(sort(other), StringRenderer()).toString)
+    }
+
+  /**
+    * 'Sort' the entries of an arbitrary upack Msg.
+    *
+    * For arrays, sorting is done by first sorting each item in the array,
+    * then comparing the 'value' of each item in the result.
+    *
+    * For objects, sorting is done by comparing the 'value' of each key in
+    * the map; values are sorted as they're added to the output.
+    *
+    * For scalars, sorting is a no-op.
+    */
+  private def sort(msg: Msg): Msg =
+    msg match {
+      case Arr(items) =>
+        val sorted = new mutable.ArrayBuffer[Msg]()
+        items.foreach(i => sorted += sort(i))
+        new Arr(sorted.sortWith(compare))
+
+      case Obj(kvs) =>
+        val sorted = new mutable.LinkedHashMap[Msg, Msg]()
+        kvs.keys.toList.map(sort).sortWith(compare).foreach(k => sorted.update(k, sort(kvs(k))))
+        Obj(sorted)
+
+      case scalar => scalar
+    }
+
+  /**
+    * Compare two arbitrary upack Msgs.
+    *
+    * If the two msgs are the same subtype, we can compare their wrapped values.
+    * If they are different subtypes, we impose a made-up ordering for consistency.
```

Comment:
Once this ordering is in place, will we get feedback from Larry on it? It looks like it should be pretty straightforward on changing the ordering if needed.

Reply:
I don't think Larry cares about the ordering itself, he just wants something in place to ensure that differences in ordering don't trigger false updates. That said, if he ever does want a change in the canonicalization logic, this should be the only place to tweak it.